src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java

```diff
@@ -22,7 +22,7 @@ class EpInfo {
   }
 
   /** Some basic verification logic to make sure the EP info is correct */
-  public void verifyEpInfo() {
+  public void verifyEpInfo(boolean enableDistinctValues) {
```

Comment:
lets add this boolean into the ctor signature instead, as its a property of the columnEps that are already saved to this object's state.

Reply:
Moved to ctor.